packages/matter.js/src/cluster/definitions/WindowCoveringCluster.ts

```diff
@@ -0,0 +1,934 @@
+         * InstalledOpenLimitTilt attribute and not smaller than InstalledClosedLimitTilt attribute. Once the command
+         * is received the TargetPositionTiltPercent100ths attribute will update its value accordingly. If the tilt
+         * value is out of bounds a response containing the status of CONSTRAINT_ERROR will be returned.
+         *
+         * @see {@link MatterApplicationClusterSpecificationV1_1} § 5.3.6.6
+         */
+        goToTiltValue: OptionalCommand(0x7, TlvGoToTiltValueRequest, 0x7, TlvNoResponse)
+    }
+});
+
+/**
+ * Window Covering
+ *
+ * The window covering cluster provides an interface for controlling and adjusting automatic window coverings such as
+ * drapery motors, automatic shades, curtains and blinds.
+ *
+ * WindowCoveringCluster supports optional features that you can enable with the WindowCoveringCluster.with() factory
+ * method.
+ *
+ * @see {@link MatterApplicationClusterSpecificationV1_1} § 5.3
+ */
+export const WindowCoveringCluster = ExtensibleCluster({
+    ...WindowCoveringBase,
+
+    /**
+     * Use this factory method to create a WindowCovering cluster with support for optional features. Include each
+     * {@link WindowCoveringFeature} you wish to support.
+     *
+     * @param features the optional features to support
+     * @returns a WindowCovering cluster with specified features enabled
+     * @throws {IllegalClusterError} if the feature combination is disallowed by the Matter specification
+     */
+    factory: <T extends `${WindowCoveringFeature}`[]>(...features: [...T]) => {
+        validateFeatureSelection(features, WindowCoveringFeature);
+        const cluster = Cluster({
+            ...WindowCoveringBase,
+            supportedFeatures: BitFlags(WindowCoveringBase.features, ...features)
+        });
+        extendCluster(cluster, TiltAndPositionAwareTiltComponent, { tilt: true, positionAwareTilt: true });
+        extendCluster(cluster, LiftComponent, { lift: true });
+        extendCluster(cluster, TiltComponent, { tilt: true });
+        extendCluster(cluster, LiftAndAbsolutePositionComponent, { lift: true, absolutePosition: true });
+        extendCluster(cluster, TiltAndAbsolutePositionComponent, { tilt: true, absolutePosition: true });
+
+        preventCluster(
+            cluster,
+            { positionAwareLift: true, lift: false },
+            { positionAwareTilt: true, tilt: false },
+            { lift: false, tilt: false }
+        );
+
+        return cluster as unknown as WindowCoveringExtension<BitFlags<typeof WindowCoveringBase.features, T>>;
+    }
+});
+
+export type WindowCoveringExtension<SF extends TypeFromPartialBitSchema<typeof WindowCoveringBase.features>> =
+    ClusterForBaseCluster<typeof WindowCoveringBase, SF>
+    & { supportedFeatures: SF }
+    & (SF extends { lift: true, positionAwareLift: true } ? typeof LiftAndPositionAwareLiftComponent : {})
```

Comment:
do we miss the lift + positionAware + absolute position ones??

Reply:
Yeah, there was a bug in the conformance parser that was dropping some components from compound expressions.  Now we've got back to a couple of ridiculously long names (hello `PinCredentialAndRfidCredentialAndFingerCredentialsNotUserComponent`)

Reply:
hihi